llvm/lib/Transforms/Utils/SSAUpdater.cpp

```diff
@@ -173,8 +173,9 @@ Value *SSAUpdater::GetValueInMiddleOfBlock(BasicBlock *BB) {
 
   // Set the DebugLoc of the inserted PHI, if available.
   DebugLoc DL;
-  if (const Instruction *I = BB->getFirstNonPHI())
-      DL = I->getDebugLoc();
+  BasicBlock::iterator It = BB->getFirstNonPHIIt();
+  if (It != BB->end())
+    DL = It->getDebugLoc();
```

Comment:
Possible alternative, YMMV:
```suggestion
  if (BasicBlock::iterator It = BB->getFirstNonPHIIt(); It != BB->end())
    DL = It->getDebugLoc();
```